src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderChunk.java

```diff
@@ -56,12 +56,11 @@
     @Shadow @Final public World world;
     @Shadow @Final private BlockPos.MutableBlockPos position;
     
-    /** Warning! Field mixed across mixins: 
+    /** Warning! Following 3 field mixed across mixins: 
      * {@link cubicchunks.asm.mixin.selectable.client.MixinRenderChunk_OptifineSpecific} and this. */
     private Cube[] cubeCache;
-    /** Warning! Field mixed across mixins: 
-     * {@link cubicchunks.asm.mixin.selectable.client.MixinRenderChunk_OptifineSpecific} and this. */
     private Chunk[] chunkCache;
+    private boolean cacheOutdated = true;
```

Comment:
What does it actually change?

Reply:
It seems that `setNeedsUpdate()` can be called from other thread (NetworkThread, I suppose). In that case if `hasEntities()` called and in a middle of check `setNeedsUpdate()` called from other thread, Java will throw NullPointerException. This commit fix that.

Reply:
Wait a moment... since when RenderChunk has it's own chunk cache?

Reply:
Only Optifine version of RenderChunk cache single chunk. I don't know since when sp614x add it.

Reply:
And how does vanilla/optifine deal with the issue, and why it's also there in the vanilla mixin?

Reply:
In vanilla those caches are not allocated and `hasEntities()` never called. It was added to common mixin because Mixin raise an error on missing interface function while loaded without Optifine.

Reply:
Then add that interface only when optifine is there?

Reply:
We can do that? Ok.